code/game/machinery/computer/card.dm

```diff
@@ -283,36 +283,53 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 		dat += " || Confirm Identity: "
 		var/S
 		var/list/paycheck_departments = list()
+		var/accessible_card = FALSE
 		if(inserted_scan_id)
 			S = html_encode(inserted_scan_id.name)
 			//Checking all the accesses and their corresponding departments
 			if((ACCESS_HOP in inserted_scan_id.access) && ((target_dept==DEPT_GEN) || !target_dept))
 				paycheck_departments |= ACCOUNT_SRV
 				paycheck_departments |= ACCOUNT_CIV
 				paycheck_departments |= ACCOUNT_CAR //Currently no seperation between service/civillian and supply
+				accessible_card = TRUE
 			if((ACCESS_HOS in inserted_scan_id.access) && ((target_dept==DEPT_SEC) || !target_dept))
 				paycheck_departments |= ACCOUNT_SEC
+				accessible_card = TRUE
 			if((ACCESS_CMO in inserted_scan_id.access) && ((target_dept==DEPT_MED) || !target_dept))
 				paycheck_departments |= ACCOUNT_MED
+				accessible_card = TRUE
 			if((ACCESS_RD in inserted_scan_id.access) && ((target_dept==DEPT_SCI) || !target_dept))
 				paycheck_departments |= ACCOUNT_SCI
+				accessible_card = TRUE
 			if((ACCESS_CE in inserted_scan_id.access) && ((target_dept==DEPT_ENG) || !target_dept))
 				paycheck_departments |= ACCOUNT_ENG
+				accessible_card = TRUE
 		else
 			S = "--------"
 		dat += "<a href='?src=[REF(src)];choice=inserted_scan_id'>[S]</a>"
 		dat += "<table>"
-		dat += "<tr><td style='width:25%'><b>Name</b></td><td style='width:25%'><b>Job</b></td><td style='width:25%'><b>Paycheck</b></td><td style='width:25%'><b>Pay Bonus</b></td></tr>"
+		dat += "<tr><td style='width:30%'><b>Name</b></td><td style='width:20%'><b>Job</b></td><td style='width:20%'><b>Department</b></td><td style='width:15%'><b>Paycheck</b></td><td style='width:15%'><b>Pay Bonus</b></td></tr>"
 
 		for(var/A in SSeconomy.bank_accounts)
+			if(!accessible_card)
+				break // don't use `return`
+
 			var/datum/bank_account/B = A
-			if(!(B.account_department in paycheck_departments))
+			if(!(B.account_department in paycheck_departments) && !(target_dept==DEPT_ALL))
```

Comment:
What are you trying to accomplish here?

Reply:
HoP console can see other department banks while they just can't adjudt their payment.

Reply:
Add to cl